### ISSUE TITLE
Aer in combination with multiprocessing

### ISSUE THREAD

**Issue by mkiffner** [Qiskit/qiskit-aer#425]
### Informations

- **Qiskit Aer version**: 0.3.2
- **Python version**: 3.7.4
- **Operating system**: ubuntu 18.04.3

### What is the current behavior?

I encounter a problem using Aer in combination with the multiprocessing package. There is no problem as long as Aer is exclusively called from within a pool. However, a single Aer evaluation outside the pool makes it impossible to run a multi-process calculation afterwards.

### Steps to reproduce the problem

```
from qiskit import QuantumCircuit, QuantumRegister, ClassicalRegister
from qiskit import Aer

from qiskit.compiler import transpile, assemble

import numpy as np

state_simulator = Aer.get_backend('statevector_simulator')
transpiled_qc = transpile(qc, backend=state_simulator)

def evaluate_state(x):

    
    para_dic = { theta : x}
    qobj = assemble(transpiled_qc.bind_parameters(para_dic), backend=state_simulator)
    job = state_simulator.run(qobj)
    state = job.result().get_statevector(transpiled_qc)
    
    return state
```
```
from multiprocessing import Pool

theta_range = np.linspace(0, 2 * np.pi, 8,endpoint = False)

p = Pool(8)
res = p.map(evaluate_state, theta_range)
print(res[-1])
```
```
print(evaluate_state(theta_range[-1]))
```
The problem occurs if another pool is defined, the calculation does not run:
```
p = Pool(8)
res = p.map(evaluate_state, theta_range)
print(res[-1])
```

### What is the expected behavior?

The second pool evaluation should run without problems, as is the case if "function_evaluation" is a generic function not using Aer.

### Suggested solutions

Note that the problem does not occur if you
(i) remove the single evaluation outside of the pool
(ii) use the same pool ,i.e., remove "p=Pool(8)" in the last cell.

Note that in my use case the function "evaluate_state" is called by a third-party program that uses multiprocessing. I thus cannot use the workarounds indicated above.

**Comment by atilag** [comment]
Yep, this is mostly because we are using multiprocessing as well on Aer/Terra, so in the end there's a heriarchy of pools that seems to conflict eachother... Pyhton stuff.
I can think of a fix, that will be a feature for next releases that consist in letting the user disable our multiprocessing logic and use their own, as this is not the first time someone tries to use multiprocessing on top of Aer.
Thanks for reporting @mkiffner !!

**Comment by atilag** [comment]
Actually, the multiprocessing part is Terra responsability (jobs management), so I'm going to move the issue there.
